### PR TITLE
Remove duplicate IDs on post list pages.

### PR DESCRIPTION
**Pull request by peterwilsoncc**
Removes duplicate IDs generated on post list tables.

* `in-{$taxonomy}-{$term_id}` elements are modified to use `wp_unique_prefixed_id()` to ensure the select and LI have different IDs
* JS duplication of quick edit to bulk edit adds a prefix -- props @siliconforks
* `#post_parent` dropdown uses a different ID for bulk edit  -- props @siliconforks

Trac ticket: https://core.trac.wordpress.org/ticket/61014

---
**This Pull Request is for code review only. Please keep all other discussion in the Trac ticket. Do not merge this Pull Request. See [GitHub Pull Requests for Code Review](https://make.wordpress.org/core/handbook/contribute/git/github-pull-requests-for-code-review/) in the Core Handbook for more details.**

### CODE REVIEW

**Review by SergeyBiryukov — APPROVED** [review]

**Comment by peterwilsoncc** [comment]
A question that's occurred to me overnight is whether both the IDs in the walker should be renamed or just one of them? 

If a plugin is targeting an ID for either CSS or JS reasons then changing both of them will certainly break the plugin. Changing one will not. My instinct is that renaming the `li`'s ID is best as CSS will still apply to the form element, and JavaScript will still capture events. @SergeyBiryukov, what do you think?

**Comment by siliconforks** [comment]
> A question that's occurred to me overnight is whether both the IDs in the walker should be renamed or just one of them?

I think you want to rename both the IDs, to handle the case where the walker gets executed twice on the same page?

**Comment by peterwilsoncc** [comment]
> I think you want to rename both the IDs, to handle the case where the walker gets executed twice on the same page?

Excellent point. There could be alternative ways of managing this but the code would be pretty ugly.

On the trac ticket, Joe Dolson mentioned that this code may possibly need to be extended further so I will wait until I hear back before committing. The ticket is on the 6.7 milestone so your portion of the code (ie, most of it) on this ticket should make it to the next release.

File: src/js/_enqueues/admin/inline-edit-post.js [hunk 1]
```diff
@@ -128,8 +128,17 @@ window.wp = window.wp || {};
 			inlineEditPost.edit( this );
 		});
 
+		// Clone quick edit categories for the bulk editor.
+		var qeCategories = $( '#inline-edit fieldset.inline-edit-categories' );
```
Comment (azaozz) [inline src/js/_enqueues/admin/inline-edit-post.js]:
```suggestion
		var beCategories = $( '#inline-edit fieldset.inline-edit-categories' ).clone();
```

Just a tiny suggestion to remove a single use var.

File: src/js/_enqueues/admin/inline-edit-post.js [hunk 2]
```diff
@@ -128,8 +128,17 @@ window.wp = window.wp || {};
...
+		var beCategories = qeCategories.clone();
```
Comment (azaozz) [inline src/js/_enqueues/admin/inline-edit-post.js]:
```suggestion
```

As above.

**Comment by azaozz** [comment]
Looks good imho too, and maintains the way HTML IDs work at the moment as far as I see (i.e. when duplicate IDs only the first element is selected/used).

**Review by joedolson — APPROVED** [review]

**Comment by joedolson** [comment]
This change wouldn't cause a problem with the hypothetical changes we might want to make in the future (changing the label structure from implicit to explicit associations), so I don't see any barriers to this.

**Comment by peterwilsoncc** [comment]
Merged [r58894](https://core.trac.wordpress.org/changeset/58894) /  https://github.com/WordPress/wordpress-develop/commit/2061a52da3d9d6d3994b0a60b2df07037d95e3d8

**Pull request closed without merging**